osf/models/preprint.py

```diff
@@ -568,6 +568,20 @@ def csl(self):  # formats node information into CSL format for citation parsing
 
         return csl
 
+    @property
+    def is_latest_version(self):
+        versioned_guid = self.versioned_guids.first()
+        guid = versioned_guid.guid
+        last_version = guid.referent
+        if not last_version.is_published:
+            last_version = Preprint.objects.filter(
+                is_published=True,
+                id__in=guid.versions.values_list('object_id', flat=True)
+            ).order_by('-versioned_guids__version').first()
+        if not last_version:
+            return False
+        return last_version.version == self.version and self.is_published
```

Comment:
Let's make this more pythonic, return faster for easy conditions.

```python
if latest_version.is_published:
    return latest_version.version == self.version
# here goes the rest of the code
```

Reply:
As for "the rest of the code":

```python
latest_version = Preprint.objects.filter(
    is_published=True,
    id__in=guid.versions.values_list('object_id', flat=True)
).order_by('-versioned_guids__version').first()
return latest_version.version == self.version if latest_version else False
```

I believe in this case, `latest_version` is always `None`. Because the only time when `guid.referent` is not published is during initial submission. However, it might be more complicated than this. We can discuss later in check-in.